api/utils.py

```diff
@@ -218,23 +226,72 @@ def addWorkMeta(cls, work, **kwargs):
     ):
         editionWorkTitle = edition.work.title
         editionWorkAuthors = edition.work.authors
+        editionInCollection = cls.checkEditionInCollection(None, edition, dbClient)
         
         return cls.formatEdition(
-            edition, editionWorkTitle, editionWorkAuthors, records, formats, showAll=showAll, reader=reader
+            edition, editionWorkTitle, editionWorkAuthors, editionInCollection, records, formats, showAll=showAll, reader=reader
         )
-    
+
+    @classmethod
+    def checkEditionInCollection(cls, work, edition, dbClient):
+
+        collectionIDs = []
+
+        if work != None:
+            for edit in dbClient.session.query(Edition) \
+                .filter(Edition.work_id == work.id):
+                    for collectEdit in dbClient.session.query(COLLECTION_EDITIONS) \
+                        .filter(COLLECTION_EDITIONS.c.edition_id == edit.id):
+                            #Avoiding duplicate collection ids
+                            if collectEdit.collection_id not in collectionIDs:
+                                for collection in dbClient.session.query(Collection) \
+                                .filter(Collection.id == collectEdit.collection_id):
+                                    numOfItems = 0
+                                    for edit in collection.editions:
+                                        numOfItems += 1
+                                    metadataOBJ = {
+                                        'id': collection.id,
+                                        'title': collection.title,
+                                        'creator': collection.creator,
+                                        'description': collection.description,
+                                        'numberOfItems': numOfItems
+                                    }
+                                    collectionIDs.append(metadataOBJ)
+                            
+        else:
+            for collectEdit in dbClient.session.query(COLLECTION_EDITIONS) \
+                .filter(COLLECTION_EDITIONS.c.edition_id == edition.id):
+                    for collection in dbClient.session.query(Collection) \
+                        .filter(Collection.id == collectEdit.collection_id):
+                            numOfItems = 0
+                            for edit in collection.editions:
+                                numOfItems += 1
+                            metadataOBJ = {
+                                'id': collection.id,
+                                'title': collection.title,
+                                'creator': collection.creator,
+                                'description': collection.description,
+                                'numberOfItems': numOfItems
+                            }
+                            collectionIDs.append(metadataOBJ)
+
+        dbClient.closeConnection()
```

Comment:
This may be a straggler, but I would remove it to avoid weird complications if the connection gets closed unexpectedly!

Reply:
Agreed. Running the API wasn't working with the closeConnection line still there.